Zombie grenadier rework

Ilysen opened the pull request:
#### Summary
```SUMMARY: Content "The Northrop Dispatch robot now replaces zombie grenadiers."```

#### Purpose of change
Zombie grenadiers are in a strange place, intelligence-wise. It appears that they have some degree of personality, and know how to use their tools despite being undead. Additionally, their ability to somehow open all of their pouches and prime their manhacks in the span of one turn on death is silly.

#### Describe the solution
Zombie grenadiers have been replaced by the Northrop Dispatch, a new robot. Designed for crowd control in extended engagement, it serves as a mobile base for manhacks, carrying and deploying them as its onboard AI requires. It has two variants:
* The NR-031 Dispatch replaces normal grenadiers. Meant for guard duty in clear sectors, it carries 10 grenade hacks, 3 tear gas hacks, 3 flashbang hacks, and 2 C-4 hacks (which it would use, pre-Cataclysm, in emergencies.)
* The military dispatch replaces elite grenadiers. It carries lethal force and has all of the manhacks elite grenadiers previously had.

There are also some key differences in the migration to the dispatch robot.
* Dispatches don't deploy their manhacks on death, instead collapsing normally. Because the hacks are fried through a failsafe in the Dispatch, you don't get them from disassembling the wreck, instead just finding some scrap parts.
* Dispatches are robots, and so are hostile towards zombies. They are also intelligent, and will path around danger. Control laptops can also turn them to your side. ***Due to technical limitations in the original grenadier, and my lack of C++ knowledge, they can't deploy their manhacks as allies.*** If someone could give me some guidance to make that possible, I'd love to do it.
* For balance reasons, both types of dispatch can only deploy one hack per 10 turns, instead of 1 per 6 turns like the normal grenadier and 1 per 3(!) turns like the elite grenadier.

I also fixed tear gas hacks, which have been broken for a very long time.

Dispatches spawn in the same places and with the same spawn rates as old grenadiers. Because migrations don't seem to exist for monsters, grenadiers still exist in the monster pool, but do not spawn naturally anywhere. Grenadiers in existing saves have the same functions as before, and the messages I chose for the special attack to deploy kamikaze hacks still works for them.

Fixes #18218.

#### Describe alternatives you've considered
Removing grenadiers. They're such a point of contention, both lore-wise and mechanics-wise, that it might have been a valid change in of itself. I think, though, that it's a valid monster design, and that further refinement and changing its lore can make it into something better.

Retaining all stats between grenadiers and their robot variants. From the start, I wanted the dispatch to be themed around crowd control, but lethal crowd control isn't strictly necessary in a lot of scenarios, and the normal grenadier has a lot of crowd control manhacks anyway, so I thought, what the heck? Let's make a riot control one - a heavy-duty robot used when riot control bots don't work out.

#### Additional context
I based its description off of a much smaller version of the [Fabrication Machine from 9](http://pre11.deviantart.net/5677/th/pre/i/2014/101/b/3/9_by_amphibizzy-d49hgke.png), so if you don't know what to envision it as, think of it like that.

![image](https://user-images.githubusercontent.com/47678781/54220294-74c8ad00-44c7-11e9-9c79-3ea06a5efb0f.png)

Shodan14 commented:
I like the idea. Maybe zombie grenadiers could make a return as suicide units at some point in the future?

Ilysen commented:
I feel like suicide grenadiers would probably be too close to bloated zombies and zombie burners, but I can think about something in the future.

Shodan14 commented:
They would follow the same theme. I've never noticed the bloated zombie explosion actually doing any damage though.

Night-Pryanik commented:
> Due to technical limitations in the original grenadier, and my lack of C++ knowledge, they can't deploy their manhacks as allies.

You mean hacked (allied) dispatch robots deploy hacks hostile to player?

Photoloss commented:
Could you please add inactive item versions, the corresponding disassembly recipes and a crafting recipe for the relaxation hack? The standard recipe books for other hacks are ["lab_elec" and "robots"](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/recipes/recipe_others.json#L1722).

Mecares commented:
As a sidenote I would love the possibility to turn my vehicle into some kind of manhack carrier that releases swarms of manhacks upon a button press and recalls them after wrecking havoc.

Ilysen commented:
@Night-Pryanik Currently, it doesn't deploy manhacks at all if it's friendly towards you. This was also the case with the original grenadier.

@Photoloss I can add relaxation hack recipes, but I feel as though the dispatch is too large to carry around feasibly by deactivating it. I imagine it would have been loaded onto vehicles during its service time.

Photoloss commented:
> I can add relaxation hack recipes, but I feel as though the dispatch is too large to carry around feasibly by deactivating it. I imagine it would have been loaded onto vehicles during its service time.

The "broken" version is "only" 100kg same as the monster. More importantly vehicle cargo spaces **also require item form** in order to transport it. At 95l you can even fit 2 of them into a cargo space.

Ilysen commented:
Security bots don't have an inactive version; I base my assumptions off of that, that it has no easy way to deactivate it for stability purposes and remaining active in engagements.

Photoloss commented:
*I* added inactive versions for those. Except due to merge conflicts Kevin applied the changes manually and seems to have missed the skitterbot. See [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/monsters.json#L2418) for the tank drone.

I haven't had time to add one for the [newly added manhack dispenser](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/monsters/defense_bot.json#L87) either but I'm planning to do so.

Ilysen commented:
Hmm, I guess that changes things. I can put it in, then; just give me a bit and I'll toss it on.

kevingranade commented on data/json/monsters/defense_bot.json:
```diff
@@ -117,5 +117,71 @@
+    "id": "mon_dispatch",
+    "type": "MONSTER",
+    "name": "riot dispatch",
+    "description": "The Northrop Dispatch, designed for crowd control situations, carries scrap parts and explosive payloads, which it rapidly assembles into kamikaze drones depending on what its AI deems necessary.  The bright green-and-yellow paint marks a riot control variant, specializing in (but not restricted to) nonlethal force.",
```
I had missed this until now since I didn't have time to review this thoroughly.  This is not ok lore-wise in DDA, it needs to simply have a payload of drones and bombs. It's also perplexing why it is even here, as it's far, far more reasonable to just say that the hacks are stored in an assembled state on-board, but have a mechanism to non-offensively self-destruct them to prevent capture.

Ilysen commented:
Okie dokie, I changed the riot variant into a "low-force" variant, changed their names to fit the other robots, and did some tweaking of their fluff, plus reverted the changes to only affect the grenadiers specifically.

Night-Pryanik commented on data/json/snippets.json:
```diff
@@ -149,6 +149,11 @@
+  {
+    "type": "snippet",
+    "category": "flier",
+    "text": [ "This is a propoganda poster showing the Northrop Dispatch's military variant.  It depicts the iconic dark green, arachnoid dispatch, standing before a fence and facing away from the camera as blurring machines rush forward from its back towards black silhouettes menacing on the horizon.  It reads: \"WE ARE HERE TO PROTECT YOU.\"" ]
```
Propaganda.
